### DISCUSSION TITLE
What is a preferred way of installing Minio: Helm or Operator?

### DISCUSSION THREAD

**Discussion by dec5e** [minio/minio#15796] [Q&A] [answered]
I saw that Operator is a proposed installation way, for example in the [asked a year ago question](https://github.com/minio/minio/discussions/11679) or at https://helm.min.io/. The last link even suggested that Helm "will be archived after April 2021". And at the same time I see Helm chart getting updates, now Operator and Helm are evolving simultaneously.

What is the today's take on questions:
- What is a preferred way of installing Minio: Helm or Operator?
- What is the difference between Minio Helm Chart and Operator?
- What are cases when Helm might be more beneficial compared to Operator?

**Comment by ravindk89** [comment, marked as answer]
If you are building for production environments in Kubernetes, use the MinIO Operator.

If you want to add some automation to the mix, use the Kustomize templates we provide as part of the Operator repository.


Our Helm Chart for Operator just installs the Operator.

Our Helm Chart for MinIO is mainly supported by the community - we are not actively maintaining it or extending its features. It's there for users who *must* use Helm or simply prefer Helm. It has whatever features the community has built out for it.

You *should* use the Operator + Kustomize wherever possible. That is where our focus on features and development are. 

The main benefit of Helm is familiarity and usability in environments that are already optimized or restricted to Helm-only setups. Personally I find kustomize more intuitive, since I can just patch a base YAML using more YAML, instead of any special syntax.
